airbyte-cdk/python/airbyte_cdk/sources/source.py

```diff
@@ -43,8 +43,9 @@ class Source(
     BaseSource[Mapping[str, Any], Union[List[AirbyteStateMessage], MutableMapping[str, Any]], ConfiguredAirbyteCatalog],
     ABC,
 ):
-    # can be overridden to change an input state
-    def read_state(self, state_path: str) -> Union[List[AirbyteStateMessage], MutableMapping[str, Any]]:
+    # can be overridden to change an input state.
+    @classmethod
+    def read_state(cls, state_path: str) -> Union[List[AirbyteStateMessage], MutableMapping[str, Any]]:
```

Comment:
This is a breaking change. We only have singer to update if we check our current repo. However, it does not mean that all custom implementations would work... The same applies to `_emit_legacy_state_format` but this one is never re-implemented in our repo

Reply:
No objection to the breaking change, but we should make this clear in the changelog